Remove use of `BackendProperties` (BackendV1) in transpiler pipeline

ElePT opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
-->

### Summary
This PR removes the following uses of `BackendProperties` deprecated in #13719:

- backend_prop input argument in DenseLayout
- properties input argument in VF2Layout
- properties input argument in VF2PostLayout
- backend_props input argument in UnitarySynthesis
- backend_properties in PassManagerConfig
- backend_properties in generate_routing_passmanager
- backend_properties in generate_translation_passmanager
- backend_properties in Target.from_configuration 

as well as the previously deprecated input arguments of `transpile`, `generate_preset_pass_manager`.

### Details and comments
Currently on hold as it contains the changes from https://github.com/Qiskit/qiskit/pull/13706, should be rebased after it's merged.

Contributes to #13708.

ElePT committed:
Remove `BackendProperties` from transpilation pipeline, remove unit tests that depend on BackendV1.

mtreinish commented on test/python/transpiler/test_vf2_post_layout.py:
```diff
@@ -587,30 +446,6 @@ def test_no_constraints(self):
         with self.assertRaises(TranspilerError):
             empty_pass.run(circuit_to_dag(qc))
 
-    def test_no_backend_properties(self):
-        """Test we raise at runtime if no properties are provided with a coupling graph."""
-        qc = QuantumCircuit(2)
-        empty_pass = VF2PostLayout(
-            coupling_map=CouplingMap([(0, 1), (1, 2)]), strict_direction=False
-        )
-        with self.assertRaises(TranspilerError):
-            empty_pass.run(circuit_to_dag(qc))
```
Isn't this test still valid without backend properties? Shouldn't the pass fail if we provide a coupling map with no error rates to use?

ElePT replied:
I thought about removing `coupling_map` from the `VF2PostLayout` pass because of this test, because it really cannot be used without backend properties. Then I went for not erroring when only a coupling map is provided, I think I checked that it simply didn't return a solution. I am now standing somewhere between the two, the only advantage of still accepting the argument I guess is some kind of compatibility angle?

mtreinish replied:
Yeah it'd mostly be a compatibility question. From a practical PoV `VF2Layout` works with just a coupling map and for the error heuristic it uses node degree in the connectivity graph as a proxy for error (so a more connected node is avoided). For `VF2PostLayout` that is unlikely to have an impact because most coupling maps are regular lattices so the degrees won't differ that much. 

The one case I could see it making a difference is like if the interaction graph of the post routing circuit was a line you could maybe snake the line over boundary nodes in the lattice that have a smaller degree. Something like:

im graph:
```
0 - 1 - 2 - 3 -4
```

cmap:
````
0 - 1 - 2 - 3
|   |    |  |
4 - 5 - 6 - 7
|   |    |  |
8 - 9 - 10 -11
````

if you used the node degree the mapping would favor putting the line that maximized the number of nodes on 0, 3, 8, and 11 in the cmap. Whether this is good behavior for VF2PostLayout is debatable though.

ElePT replied:
Hmm ok, even though that use case doesn't come up much in practice now, I think the most "behavior preserving" path regarding this removal is erroring on the coupling map.

mtreinish commented on qiskit/transpiler/preset_passmanagers/generate_preset_pass_manager.py:
```diff
@@ -27,7 +27,7 @@
 from qiskit.transpiler.layout import Layout
 from qiskit.transpiler.passmanager_config import PassManagerConfig
-from qiskit.transpiler.target import Target, target_to_backend_properties
+from qiskit.transpiler.target import Target
```
Are we planning to remove `target_to_backend_properties` in the PR removing the class?

mtreinish reviewed (commented):
This LGTM, one inline question about a removed test but outside of that I think this is good to merge.

ElePT committed:
Merge branch 'main' of https://github.com/Qiskit/qiskit into remove-v1-props-transpiler

ElePT committed:
Remove coupling_map input in VF2PostLayout pass, as it would only be used if backend properties were present. After the removal of the latter, the coupling_map would always be overwritten by the target.

ElePT commented:
After a bit of back and forth between the different removal alternatives in `VF2PostLayout`, I settled for removing the `coupling_map` input argument. Unlike `VF2Layout`, which might use `coupling_map` in the presence of a `target` if the target is missing the connectivity information, `VF2PostLayout` had no use for it as a standalone input. The version change seems like a good opportunity to get rid of both inputs as a "pack". I updated the deprecation in #13719 to include the `coupling_map` and removed it in [fe772b9](https://github.com/Qiskit/qiskit/pull/13722/commits/fe772b970926b8119bb5119ee58917f9c8dbfde5).

mtreinish commented on qiskit/transpiler/passes/layout/vf2_post_layout.py:
```diff
@@ -153,16 +142,12 @@ def __init__(
-                "A target must be specified or a coupling map and properties must be provided"
-            )
+        if self.target is None:
+            raise TranspilerError("A target must be specified or a coupling map must be provided")
```
This threw me for a second until I remember these passes should be instantiable even if the required target isn't specified.

ElePT replied:
Yes I also spent a long time thinking if I should make it a required argument.

mtreinish reviewed (approved):
This LGTM now thanks for doing all of the work on this.
